scripts/zephyr_module.py

```diff
@@ -178,10 +178,13 @@ def main():
                         help='List of extra modules to parse')
     parser.add_argument('-w', '--west-path', default='west',
                         help='Path to west executable')
+    parser.add_argument('-z', '--zephyr-base',
```

Comment:
I would actually like to deprecate this option as we move away from west knowing specifically about this variable. Is there another way to do this, e.g. with the environment?

Reply:
How will you ensure `west build` works out-of-tree then ?

Reply:
The same way we do today? Or by having each extension responsible for discovering ZEPHYR_BASE on its own?

Reply:
today we use ZEPHYR_BASE for out of tree builds.
and how will you execute an extension command if you do not know where the project is located ?
As example, user has following structure:
```
/projects/zephyr-project/zephyr
/projects/zephyr-project/.west
/projects/zephyr-project/modules/...
/projects/zephyr-project/...
/projects/my-oot-project
```
then he does:
```
$ cd /projects/my-oot-project
$ west build -b<board>
```
in this set-up, `west` uses ZEPHYR_BASE to find:
location of Zephyr, and from there west topdir, and then it can load the extension command.

As seen here:
```
$ cd /projects/oot/hello_world
$ west build -bnrf52840_pca10056
usage: west [-h] [-z ZEPHYR_BASE] [-v] [-V] <command> ...
west: error: argument <command>: invalid choice: 'build' (choose from 'init', 'update', 'list', 'manifest', 'diff', 'status', 'forall', 'help', 'config', 'topdir', 'selfupdate')
$ 
```

Reply:
one feature I would like in future `west` is the ability to specify any west config option on command line on the fly, as example:
`west -c myconf.setting=value <command>`
that way user could do:
`west -c zephyr.base=/projects/ncs/zephyr list`

and `west does not need to know anything about the meaning of `zephyr.base`, it just adds it to its internal config values, in exact same manner as if user had done:
```
west config zephyr.base /projects/ncs/zephyr
west list
```
except for the difference that a `-c myconf.setting=value` is not stored persistently.
That would be nice for scripting.

Reply:
> today we use ZEPHYR_BASE for out of tree builds.
> and how will you execute an extension command if you do not know where the project is located ?

This all works today without using the --zephyr-base option. It uses the ZEPHYR_BASE environment variable, true, but changing that to the option isn't really any better IMO, especially given we are trying to get rid of zephyr specifics in the core west, and just pushing those out to extensions.

Can't this command just be run with ZEPHYR_BASE in the environment, so the existing hacks in west continue to be used, instead of using --zephyr-base, which I believe is otherwise not used in the zephyr build system?

Reply:
actually, this line you are commenting, is not `west` code, it's a Zephyr script for looking up modules.
If you take a look at those lines:
https://github.com/zephyrproject-rtos/zephyr/blob/b9d3ed510ccbd7698e1b758e3a7730ad059ffcf1/scripts/zephyr_module.py#L186-L189

you'll notice that the zephyr base is translated into `cwd` when calling `west`.
So actually I assume you should be fine current solution.

Reply:
> actually, this line you are commenting, is not `west` code, it's a Zephyr script for looking up modules.

Obviously :P

When I wrote:

> I would actually like to deprecate this option 

I meant deprecate the option *in west itself*

Reply:
> When I wrote:
> 
> > I would actually like to deprecate this option
> 
> I meant deprecate the option _in west itself_

Of course, and you can still do, cause this code doesn't use the `west` option.
But your comment did get my thinking side-tracked for a while :stuck_out_tongue:

Reply:
D'oh!!